sql: support SHOW COLUMNS ... WITH COMMENTS

hueypark opened the pull request:
Informs #36439.

Release note (sql change): support SHOW COLUMNS ... WITH COMMENTS
now supports printing out column comments using the optional phrase
`WITH COMMENT`, e.g `SHOW COLUMNS FROM mytable WITH COMMENT`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/37174)
<!-- Reviewable:end -->

jordanlewis commented:
@hueypark thanks for the contribution and sorry about the delay in review. Could you please rebase this one?

hueypark commented:
@jordanlewis Of course, I'll do it before the weekend.

hueypark committed:
sql: support SHOW COLUMNS ... WITH COMMENTS

Release note (sql change): support SHOW COLUMNS ... WITH COMMENTS
now supports printing out column comments using the optional phrase
`WITH COMMENT`, e.g `SHOW COLUMNS FROM mytable WITH COMMENT`.

jordanlewis commented:
Thanks!

bors r+
